### ISSUE TITLE
Aquired time randomly switch to datetime

### ISSUE THREAD

**Issue by OBVYamaska** [opengisch/QField#3635]
### Describe the issue
While importing data from these last months projects, I came across a weird bug that seem to happen randomly. I never had the problem on other projects, but it happened at different dates, on different projects this fall.
To put it simply: sometimes, QField will save a datetime format instead of a simple time format. All in all, it might not seem like such a problem, but I realized it caused my imports to fail mi-process... leaving me with random amount of data that did sync, and other that didn't. Quite a puzzle to correct manually and a lot of time lost!

As you can see here, on the same project, after more than 200 entities created, one switched to datetime... and went back to normal after that.
![image](https://user-images.githubusercontent.com/101751953/201780909-20ba021f-d996-48b2-8cf2-c5f4cd4eeff2.png)

This is how the field is configured:
![image](https://user-images.githubusercontent.com/101751953/201781207-da0f53fd-acb7-4184-a01d-e7443ed62447.png)

### Reproduction steps
1. [Go to '...']
2. [Click on '...']
3. [Scroll down to '...']

### Desktop (please complete the following information)
It happened with different devices (QField) and at least two versions of QGIS (the problem doesn't seem to be on the desktop side though)

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: I just realized the problem, but found it on datas captured from 2022-06-01 to 2022-10-30. We keep our devices up to date with QField versions...
* Problem can be reliably reproduced, doesn't happen randomly: [No]
* Problem happens with all files and projects, not only some files or projects: [No]

**Comment by nirvn** [comment]
@OBVYamaska , OK, I've looked into this a bit, and time-only handling within QField's date/time editor widget is essentially not properly implemented yet (I've updated the description accordingly).

That being said, you can get away with the current state of affairs by changing two things:
a/ use this expression as your default value: `substr(to_string(to_time(now())),1,8)` 
b/ in the field format, change time to custom, and use the following custom string: `hh:mm:ss` (capital HH isn't supported in QML)

That should provide a reliable way to add time values for the time being.

**Comment by OBVYamaska** [comment]
> @OBVYamaska , OK, I've looked into this a bit, and time-only handling within QField's date/time editor widget is essentially not properly implemented yet (I've updated the description accordingly).
> 
> That being said, you can get away with the current state of affairs by changing two things: a/ use this expression as your default value: `substr(to_string(to_time(now())),1,8)` b/ in the field format, change time to custom, and use the following custom string: `hh:mm:ss` (capital HH isn't supported in QML)
> 
> That should provide a reliable way to add time values for the time being.

Good idea. I did suspect this problem at first so I didn't set proper boundaries for the default value, but this will indeed solve the problem for the time being.

**Comment by m-kuhn** [comment]
Let's keep it open. Maybe someone looking for a proper implementation wants to invest into this. It should be relatively simple.

**Comment by nirvn** [comment]
@OBVYamaska , have things improved on that front with latest QField 3.7? We revamped the datetime editor widget back in 3.6.
